code/hash/n-merge-bang-vs-[]=.rb

```diff
@@ -0,0 +1,22 @@
+require 'benchmark/ips'
+
+ENUM = (1..100)
+HASH = ENUM.to_h{ |a| [a, a]}
+SECOND_HASH = ENUM.to_h{ |a| [a+50, a+50]}
+
+def merge
+  h = HASH
+  h.merge!(SECOND_HASH)
+end
+
+def assign
+  ENUM.each_with_object({}) do |e, h|
```

Comment:
Function `merge` works with a hash prefilled with 100 entries, while `assign` works with a blank new hash.
```suggestion
  ENUM.each_with_object(SECOND_HASH) do |e, h|
```

Reply:
P.S. though my suggestion is [`ENUM` ➡ `SECOND_HASH`] while `merge` is [`SECOND_HASH` ➡ hash version of `ENUM`]